File: R/CPPSAS7BDAT/R/read_sas.R
#
# @brief: Wrapper for the C++ classes
# @author: Olivia Quinet
#

read_sas <- function(filename, include=NULL, exclude=NULL)
{
  sink <- SinkWholeData$new();
  r <- sas_reader(filename, sink, include=include, exclude=exclude);
  r$read_all();
  sink;
}

File: python/pycppsas7bdat/read_sas.py
# -*- coding: utf-8 -*-
# @brief: Wrapper for the C++ classes 
# @author: Olivia Quinet
#

from pycppsas7bdat import Reader
from pycppsas7bdat.sink import SinkWholeData

def read_sas(inputfilename, sink=None, include=None, exclude=None):
    if sink is None: sink = SinkWholeData()
    reader = Reader(inputfilename, sink, include=include, exclude=exclude)
    reader.read_all()
    return sink
